Library/Homebrew/language/go.rb

```diff
@@ -7,7 +7,11 @@ module Go
     # The resource names should be the import name of the package,
     # e.g. `resource "github.com/foo/bar"`
     def self.stage_deps(resources, target)
-      opoo "tried to stage empty resources array" if resources.empty?
+      if resources.empty? && ENV["HOMEBREW_DEVELOPER"].nil?
+        opoo "tried to stage empty Language::Go resources array"
+      elsif resources.empty?
+        odie "tried to stage empty Language::Go resources array"
+      end
```

Comment:
Let's add some nesting to avoid repeating the condition (which makes the code less obvious in my opinion):

```ruby
if resources.empty?
  if ARGV.homebrew_developer?
    odie "tried to stage empty Language::Go resources array"
  else
    opoo "tried to stage empty Language::Go resources array"
  end
end
```

This duplication makes me want to have a `opoo_for_mere_mortals_and_odie_for_developers` method (obviously with a better name than this). Maybe something like this (not available today), would make sense:

```ruby
opoo "tried to stage empty Language::Go resources array",
     :fatal => ARGV.homebrew_developer?
```